### PR TITLE
ACA-121 - Remove use of elastic search when fetching cases

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ACA-121

### Change description ###

Use the standard SQL based get case endpoint rather than the Elastic Search API to get the case for which Access is being assigned.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java [hunk 1]
```diff
@@ -195,8 +197,16 @@ private UserDetails toUserDetails(ProfessionalUser user, List<String> caseRoles)
     }
 
     private CaseDetails getCase(CaseAssignment input) {
-        Optional<CaseDetails> caseOptional = dataStoreRepository.findCaseBy(input.getCaseTypeId(), input.getCaseId());
-        return caseOptional.orElseThrow(() -> new CaseCouldNotBeFetchedException(CASE_COULD_NOT_BE_FETCHED));
+        CaseDetails caseDetails = null;
+        try {
+            caseDetails = dataStoreRepository.findCaseByCaseIdExternalApi(input.getCaseId());
+        } catch (FeignException e) {
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java:203]: Can this Feign block inside repository ??

File: src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/DataStoreApiClient.java [hunk 2]
```diff
@@ -33,4 +35,7 @@ CaseUserRoleResource getCaseAssignments(@RequestParam("case_ids") List<String> c
     @DeleteMapping(value = CASE_USERS, consumes = APPLICATION_JSON_VALUE)
     void removeCaseUserRoles(@RequestBody CaseUserRolesRequest userRolesRequest);
 
+    @GetMapping(CASES_WITH_ID)
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/DataStoreApiClient.java:38]: What is stopping us to remove ES api completely "searchCases" ? 
(also from wiremock)

File: src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java (hunk 1 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java]: Resolved in https://github.com/hmcts/aac-manage-case-assignment/pull/173/commits/d691c210e88b0c5d05681a6b8fcd5b123858c829

File: src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/DataStoreApiClient.java (hunk 2 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/DataStoreApiClient.java:34]: Resolved in https://github.com/hmcts/aac-manage-case-assignment/pull/173/commits/2fd9ff1c23b960edebd4606cfb08c91cccb83ce4

**Review by smathangi — APPROVED** [review]
LGTM

**Pull request merged**